packages/actor-http-fetch/lib/ActorHttpFetch.ts

```diff
@@ -76,12 +76,46 @@ export class ActorHttpFetch extends ActorHttp {
 
     try {
       requestInit = await this.fetchInitPreprocessor.handle(requestInit);
+      let retryCount: number = action.context?.get(KeysHttp.httpRetryCount) ?? 0;
 
-      // Perform request
       const customFetch: ((input: RequestInfo, init?: RequestInit) => Promise<Response>) | undefined = action
         .context?.get(KeysHttp.fetch);
 
-      const response = await (customFetch || fetch)(action.input, requestInit);
+      // Actual fetch respecting timeout and retry count.
+      const getResponse = async(): Promise<Response> => {
+        // When retry count is greater than 0, record the errors for failed requests.
+        const errors: unknown[] = [];
+        while (retryCount-- >= 0) {
+          try {
```

Comment:
Open question: next to catching network errors, should we also handle server errors (probably only those in the 500 range then)?

Reply:
By default, probably not, but an option for this might be valuable.

Reply:
added an option httpRetryOn5xx :)

Reply:
Nice!